src/args.c

```diff
@@ -169,6 +179,21 @@ int ashuffle_options(struct ashuffle_options * opts,
                 return -1;
             }
             state = NO_STATE;
+        } else if (state == HOST) {
+            opts->host = malloc(strlen(argv[i]) + 1);
+            if (opts->host == NULL) {
+                fputs("Error processing host\n", stderr);
+                return -1;
+            }
+            snprintf(opts->host, strlen(argv[i]) + 1, "%s", argv[i]);
```

Comment:
snprintf is a complicated tool for a string copy. Just use a `memcpy` here:

```
memcpy(opts->host, argv[i], strlen(argv[i]) + 1);
```

Even better, you could just use `strdup` here, so you don't have to muck about with string lengths.

Reply:
fixed in 9fa876d